### PR TITLE
Add xscen functions and move to conda environment

### LINKED ISSUE #19: Importations xscen
### Addressing a Problem?

`xscen` possède déjà plusieurs fonctionnalités qui pourraient être réutilisées directement dans `xhydro`, notamment pour le calcul d'indicateurs (excluant les indicateurs fréquentiels plus avancés) et pour les fonctions nécessaires aux analyses hydroclimatiques (`climatological_mean`, `compute_deltas`, `ensemble_stats`, `generate_weights`, `produce_horizon`).

### Potential Solution

Plutôt que de copier-coller le code, je propose d'importer les fonctions `xscen` pertinentes et de les passer au `__init__`, ce qui signifie qu'on peut par exemple directement utiliser `xhydro.ensemble_stats`, sans avoir à se soucier du fait que le code se trouve dans une autre librairie. Il existe également la mécanique pour transposer la documentation vers le ReadTheDocs.

### Additional context

J'ai créé une branche, pour que vous puissiez voir à quoi ça pourrait ressembler: https://github.com/hydrologie/xhydro/tree/indicators/xhydro 

### Contribution

- [X] I would be willing/able to open a Pull Request to contribute this feature.

### PR DESCRIPTION
**Pull request by RondeauG**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
  - This PR fixes #19
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features).
- [x] (If applicable) Tests have been added.
- [x] HISTORY.rst has been updated (with summary of main changes).
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added.

### What kind of change does this PR introduce?

* Adds a few `xscen` functions to `xhydro`, mainly related to the computation of indicators and hydroclimatic statistics.

### Does this PR introduce a breaking change?

- Deleted `xhydro.py`. I don't think that we'll ever need this file, but it can be readded later.

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -3,3 +3,4 @@ channels:
   - conda-forge
 dependencies:
   - python >=3.9,<3.11
```
Comment (Zeitsperre) [inline environment.yml:5]:
```suggestion
  - python >=3.9,<3.12
```

**Comment by Zeitsperre** [comment]
The failures are due to a packaging issue with xESMF. Let me see if I can find the discussion...

File: docs/conf.py [hunk 2]
```diff
@@ -21,16 +21,14 @@
...
-import xhydro  # noqa
-
-
 if os.environ.get("READTHEDOCS") and "ESMFMKFILE" not in os.environ:
     # RTD doesn't activate the env, and esmpy depends on an env var set there
     # We assume the `os` package is in {ENV}/lib/pythonX.X/os.py
     # See conda-forge/esmf-feedstock#91 and readthedocs/readthedocs.org#4067
     os.environ["ESMFMKFILE"] = str(Path(os.__file__).parent.parent / "esmf.mk")
 
+import xhydro  # noqa
```
Comment (Zeitsperre) [inline docs/conf.py:30]: Oof! My bad!

**Comment by RondeauG** [comment]
If that's fine with you, I'd wait for #27 for the documentation, since we'll be able to leverage the notebook that has been started there.

File: xhydro/cc.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+# FIXME: To be deleted once climatological_op is available in xscen
+def climatological_op(ds, **kwargs):
+    """Compute climatological operation.
+
+    Notes
+    -----
+    This is a temporary wrapper to be deleted once climatological_op is available in xscen.
+    For the time being, it is a simple wrapper around climatological_mean.
+    See :py:func:`xscen.climatological_mean` for more details.
```
Comment (RondeauG) [inline xhydro/cc.py:25]: @Zeitsperre This does not seem to link to the xscen documentation once on RtD... Any idea why?

Comment (Zeitsperre) [reply xhydro/cc.py:25]: `xscen` is not within the scope of the project. This is a case for intersphinx, so it needs to be called differently.

Comment (Zeitsperre) [reply xhydro/cc.py:25]: hmm oh I see. This is strange.

Comment (RondeauG) [reply xhydro/cc.py]: Looks like we can't just shortcut the link! This should be fixed now.

**Pull request merged**